### PR TITLE
Replacing jQuery with native methods

### PR DESCRIPTION
**Pull request by sivakumar-kailasam**
Many apps decide to exclude jQuery and it makes it impossible if addons use them. Its good that material components themselves don't care about jQuery. This is PR removes all dependency on jQuery.

### CODE REVIEW

**Comment by sivakumar-kailasam** [comment]
I've raised a PR to https://github.com/chrislopresto/ember-freestyle/pull/157 that removes freestyle's internal dependency on jQuery. Once it lands, it'd be safe to update to that version while this PR uses the branch of that PR.

**Comment by sivakumar-kailasam** [comment]
@Kerrick the previous failure was because of a lack of changes on .travis.yml. Its fixed now.

**Comment by Kerrick** [comment]
Thanks for the PR! Definitely in favor of this strategy. Once code review has passed we can merge & ship this 🎉

File: addon/components/mdc-drawer.js [hunk 1]
```diff
@@ -109,8 +111,8 @@ export default Component.extend(MDCComponent, {
       notifyOpen: () => set(this, 'open', true),
       notifyClose: () => set(this, 'open', false),
       isRtl: () => getElementProperty(this, 'direction') === 'rtl',
-      getFocusableElements: () => this.$(FOCUSABLE_ELEMENTS),
-      hasNecessaryDom: () => !!get(this, 'element') && !!this.$(ITEMS_SELECTOR).length,
+      getFocusableElements: () => this.element.querySelectorAll(FOCUSABLE_ELEMENTS),
+      hasNecessaryDom: () => Boolean(this.element),
```
[the remark this answers is not in the document]
Comment (sivakumar-kailasam) [reply addon/components/mdc-drawer.js]: This was the check in their default adapter, so went with it. Also we're guaranteed that the necessary dom exists if the element gets rendered in this case.

**Comment by Kerrick** [comment]
Looks like the only thing left is for https://github.com/chrislopresto/ember-freestyle/pull/157 to get updated so this PR's branch can point to a published version of ember-freestyle.

**Comment by sivakumar-kailasam** [comment]
@Kerrick this is good to merge now.

**Comment by danielraggs** [comment]
@sivakumar-kailasam Your PR is almost good to go; however, I did some quick testing and found that the MDC-menu component has a bug in your branch: once the menu is opened, you're not able to click outside to close the menu. After you fix this bug your PR will be good to merge! 👍

**Comment by sivakumar-kailasam** [comment]
@danielraggs Sure will take a look at that and see if I can add a test case for it while take a stab at it 😄

**Comment by sivakumar-kailasam** [comment]
@danielraggs added a fix but couldn't add a test for it yet. I hope to add acceptance tests when I raise a different PR to update the deps.

**Pull request merged**